### PR TITLE
Fixes Voter Widget width on campaign pages

### PR DESCRIPTION
**Pull request by aaronschachter**
### What's this PR do?

This pull request adds a check for the current URL to determine whether a `CivicEngineVoterWidget` is being rendered on a campaign page. If it is, render the parent `ContentBlock` at 2/3 instead of full width. Fixes a bug introduced in #2382 

Campaign page

### How should this be reviewed?

👀 

### Any background context you want to provide?

I'm a little stuck on why the test for rendering a `GetOutTheVoteBlock` is failing:

> Error: Uncaught [Invariant Violation: Could not find "client" in the context or passed in as an option. Wrap the root component in an <ApolloProvider>, or pass an ApolloClient instance in via options.]

I've looked in https://github.com/apollographql/react-apollo/issues/2900 a bit but no luck, our versions seem okay. Another error occurs too - coming from the `GetOutTheVoteBlock`. I don't see this error though when inspecting the console on [a campaign page](https://qa.dosomething.org/us/campaigns/your-plan-your-vote/action).

>   Warning: Failed prop type: You provided a `checked` prop to a form field without an `onChange` handler. This will render a read-only field. If the field should be mutable use `defaultChecked`. Otherwise, set either `onChange` or `readOnly`.

Because this bug is blocking a deploy, I think it's fine to leave fixing the test for later.

### Relevant tickets

References [Pivotal #]().

### Checklist

- [ ] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added screenshots of front-end changes on small, medium, and large screens.
- [ ] Added appropriate feature/unit tests.

### CODE REVIEW

**Comment by aaronschachter** [comment]
Review app links:
* [Story page](https://dosomething-phoenix-de-pr-2386.herokuapp.com/us/stories/voter-widget) -- full width
* [Campaign page](https://dosomething-phoenix-de-pr-2386.herokuapp.com/us/campaigns/test-groups-campaign/voter-widget) -- 2/3 (must be authenticated)

File: resources/assets/components/blocks/ContentBlock/ContentBlock.test.js [hunk 1]
```diff
@@ -68,11 +68,58 @@ describe('ContentBlock component', () => {
       );
     });
 
+    /** @test */
+    /*
+    // @TODO: Fix this failing test: Invariant Violation: Could not find "client" in the context or passed in as an option. Wrap the root component in an <ApolloProvider>, or pass an ApolloClient instance in via options.
```
Comment (mendelB) [inline resources/assets/components/blocks/ContentBlock/ContentBlock.test.js:73]: This is probably coming from the [GraphQL query](https://github.com/DoSomething/phoenix-next/blob/51d8409e3e2e10088d2d0b7cffd81aef69c87a29/resources/assets/components/utilities/Embed/Embed.js#L61) we're making in the [rendered `Embed` components](https://github.com/DoSomething/phoenix-next/blob/51d8409e3e2e10088d2d0b7cffd81aef69c87a29/resources/assets/components/blocks/GetOutTheVoteBlock/GetOutTheVoteBlock.js#L48-L52).

We might be able to get away with this using `shallow` to render the component, but with the transition to react testing libraries `render`, I think we need to wrap this in a `MockedProvider` like we do [here](https://github.com/DoSomething/phoenix-next/blob/51d8409e3e2e10088d2d0b7cffd81aef69c87a29/resources/assets/components/pages/AccountPage/Credits/VolunteerCreditsQuery.test.js#L46).

File: resources/assets/components/blocks/ContentBlock/ContentBlock.test.js [hunk 2]
```diff
@@ -68,11 +68,58 @@ describe('ContentBlock component', () => {
...
+        'col-span-3',
+      );
+    });
+
+    /** @test */
+    test('displays across two-thirds of the row if footerType is CivicEngineVoterWidget and viewed on a campaign page', () => {
+      delete window.location;
+      window.location = new URL(
+        'https://dev.dosomething.org/us/campaigns/test-campaign',
+      );
```
Comment (mendelB) [inline resources/assets/components/blocks/ContentBlock/ContentBlock.test.js:102]: I'm not positive that this is better practice, but I've been interacting with the `jsdom` to reconfigure the `location` (is that even the context for react testing library tests?)

https://github.com/DoSomething/phoenix-next/blob/51d8409e3e2e10088d2d0b7cffd81aef69c87a29/resources/assets/helpers/refer-friends.test.js#L30

File: resources/assets/components/blocks/ContentBlock/ContentBlock.js [hunk 3]
```diff
@@ -37,6 +37,11 @@ const ContentBlock = ({
 }) => {
   const contentNode = content ? <TextContent>{content}</TextContent> : null;
   const { footerType } = additionalContent;
+  const fullWidthFooter =
+    footerType === 'GetOutTheVoteBlock' ||
+    // Only display CivicEngineVoterWidget as full width when not rendered in a campaign page.
```
Comment (mendelB) [inline resources/assets/components/blocks/ContentBlock/ContentBlock.js:42]: I've found it helpful (or maybe just plain gotten a kick out of) when we sort of call out the `Hack:` in the comments formally, e.g. https://github.com/DoSomething/phoenix-next/blob/51d8409e3e2e10088d2d0b7cffd81aef69c87a29/webpack.config.js#L39

**Review by mendelB — APPROVED** [review]
📏

**Pull request merged**